src/main/java/org/testeditor/fixture/core/AbstractTestCase.java

```diff
@@ -12,35 +12,48 @@
  *******************************************************************************/
 package org.testeditor.fixture.core;
 
-import java.util.concurrent.TimeUnit;
-
 import org.junit.After;
 import org.junit.Before;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
+import org.testeditor.fixture.core.TestRunReporter.SemanticUnit;
 
+/**
+ * Class from which all generated unit tests are (transitively) derived
+ */
 public class AbstractTestCase {
 
-	protected static final Logger logger = LoggerFactory.getLogger(AbstractTestCase.class);
-	private long start;
+	protected final TestRunReporter reporter;
+	private final TestRunListener logListener;
+	
+	public AbstractTestCase() {
+		// initialization is done in ctor to allow other ctors to access reporter
+		// to allow registration before the first event is reported (ENTER TEST)
+		reporter=createTestRunReporter();
+		logListener=createLoggingListener();
+	}
 
 	@Before
 	public void initTestLaunch() {
-		MDC.put("TestName", "TE-Test: " + this.getClass().getSimpleName());
-		logger.info("****************************************************");
-		logger.info("Running test for {}", this.getClass().getName());
-		start = System.currentTimeMillis();
-		logger.info("****************************************************");
+		MDC.put("TestName", "TE-Test: " + getClass().getSimpleName());
```

Comment:
I see that this line was in here before, but why do we need to add "TE-Test: " to the context? If it's used for parsing the log it should go into the logging configuration.